pkg/test-fw/ig/ig.go

```diff
@@ -0,0 +1,273 @@
+func (ig *IG) kill() error {
+	const sig syscall.Signal = syscall.SIGKILL
+
+	// No need to kill, command has not been executed yet or it already exited
+	if ig.cmd == nil || (ig.cmd.ProcessState != nil && ig.cmd.ProcessState.Exited()) {
+		return nil
+	}
+
+	// Given that we set Setpgid, here we just need to send the PID of ig.cmd
+	// (which is the same PGID) as a negative number to syscall.Kill(). As a
+	// result, the signal will be received by all the processes with such PGID,
+	// in our case, the process of ig.cmd.
+	err := syscall.Kill(-ig.cmd.Process.Pid, sig)
+	if err != nil {
+		return err
+	}
+
+	// In some cases, we do not have to wait here because the cmd was executed
+	// with run(), which already waits. On the contrary, in the case it was
+	// executed with start() thus ig.started is true, we need to wait indeed.
+	if ig.started {
+		err = ig.cmd.Wait()
+		if err == nil {
+			return nil
+		}
+
+		// Verify if the error is about the signal we just sent. In that case,
+		// do not return error, it is what we were expecting.
+		var exiterr *exec.ExitError
+		if ok := errors.As(err, &exiterr); !ok {
+			return err
+		}
+
+		waitStatus, ok := exiterr.Sys().(syscall.WaitStatus)
+		if !ok {
+			return err
+		}
+
+		if waitStatus.Signal() != sig {
+			return err
+		}
+
+		return nil
+	}
+
+	return err
+}
+
+func (ig *IG) stop() error {
+	if !ig.started {
+		return fmt.Errorf("Warn(%v): trying to stop command but it was not started", ig.cmd)
+	}
+
+	if err := ig.kill(); err != nil {
+		return err
+	}
+
+	ig.started = false
+	return nil
+}
+
+type option func(*IG)
+
+func WithPath(path string) option {
+	return func(ig *IG) {
+		ig.path = path
+	}
+}
+
+func WithImage(image string) option {
+	return func(ig *IG) {
+		ig.image = image
+	}
+}
+
+// Flags should be in the form: "--flag_name=value" or "-shorthand=value"
+func WithFlags(flags ...string) option {
+	return func(ig *IG) {
+		ig.flags = flags
+	}
+}
+
+// Elements of env should be in form: "ENV_VARIABLE_NAME=value"
+func WithEnv(env ...string) option {
+	return func(ig *IG) {
+		ig.env = env
+	}
+}
+
+func WithStartAndStop() option {
+	return func(ig *IG) {
+		ig.StartAndStop = true
+	}
+}
+
+// Runs "ig version" to get the version string
+func getIgVersionString(path string) (string, error) {
+	cmd := exec.Command(path, "version")
+
+	var out bytes.Buffer
+	cmd.Stdout = &out
+
+	if err := cmd.Run(); err != nil {
+		return "", err
+	}
+	return out.String(), nil
+}
+
+func extractIgVersion(str string) (semver.Version, error) {
+	parsedVersion, err := semver.ParseTolerant(str)
+	if err != nil {
+		return parsedVersion, fmt.Errorf("parsing version from string[%s]: %w]", str, err)
+	}
+	return parsedVersion, nil
+}
+
+// New creates a new IG configured with the options passed as parameters.
+// Supported parameters are:
+//
+//	WithImage(gadget_image)
+//	WithPath(string)
+//	WithEnv(...string)
+//	WithFlags(...string)
+//	WithStartAndStop()
+func New(opts ...option) (*IG, error) {
```

Comment:
As this is only designed for testing purposes, I think this method should take a testing.T object and call t.Fail (or similar) instead of returning errors. (This will remove a lot of boilerplate error handling code around).